main/swadge2024.h

```diff
@@ -349,6 +353,17 @@ typedef struct
      * globalMidiPlayerFillBuffer() will be used instead to fill sample buffers
      */
     fnDacCallback_t fnDacCb;
+
+    /**
+     * @brief This function is called to fill in a SwadgePass packet with mode-specific data. The Swadge mode should
+     * only fill in it's relevant data and not touch other mode's data.
```

Comment:
Is this just going to be one struct that every mode shares and adds its own members to as needed?

Reply:
Yep! The alternative would be giving letting each mode append to a pointer before transmission, then letting each mode strip off bytes from a pointer on reception. That feels way more error prone though, since it requires symmetry in the write & read functions for each mode.